### PR TITLE
Fix healthcheck kwargs for container.create()

### PR DESCRIPTION
**Pull request by inknos**
The health_* kwargs were being passed as snake_case in the JSON payload, but /libpod/containers/create endpoint expects nested objects.

Assemble kwargs into proper structures:
- healthconfig: {Test, Interval, Timeout, Retries, StartPeriod}
- startupHealthConfig: {Test, Interval, Timeout, Retries, Successes}
- healthLogDestination, healthMaxLogCount, healthMaxLogSize (camelCase)
- health_on_failure string mapped to health_check_on_failure_action int

Also, it's possible to pass duration values as go-style strings ("30s", "1m", "500ms") and integers as nanoseconds.

Fixes: https://redhat.atlassian.net/browse/RHEL-140716

---

keeping as draft since I want to sleep on the code before saying it's ready

### CODE REVIEW

**Review by giuseppe — APPROVED** [review]
LGTM

File: podman/api/parse_utils.py [hunk 1]
```diff
@@ -11,6 +12,65 @@
...
+    pos = 0
+    while pos < len(remaining):
+        match = pattern.match(remaining, pos)
+        if not match:
+            raise ValueError(
+                f"Invalid duration format: {value!r}. "
+                f"Use Go-style durations like '30s', '1m', '500ms', '1h30m'."
+            )
+        num_str, unit = match.groups()
+        total_ns += int(float(num_str) * DURATION_UNITS_NS[unit])
```
Comment (Honny1) [inline podman/api/parse_utils.py]: Well, using `float` feels odd. Why not `int`?

Comment (Honny1) [reply podman/api/parse_utils.py]: (`"1"` -> `1.0000000001` could happen)

File: podman/api/parse_utils.py [hunk 2]
```diff
@@ -11,6 +12,65 @@
...
+    "none": 0,
+    # value 1 is "invalid" in the Go enum (HealthCheckOnFailureActionInvalid)
+    # and should not be used directly
+    "kill": 2,
+    "restart": 3,
+    "stop": 4,
+}
+
+
+def prepare_duration_ns(value: Union[int, str, None]) -> Union[int, None]:
```
Comment (Honny1) [inline podman/api/parse_utils.py:35]: What about `prepare_duration_ns("0")`?

File: podman/domain/containers_create.py [hunk 3]
```diff
@@ -823,15 +835,90 @@ def parse_host_port(_container_port, _protocol, _host):
...
-            params["no_healthcheck"] = args.pop("no_healthcheck", None)
-        else:
-            for hc in set(args) & health_commands_data:
-                params[hc] = args.pop(hc, None)
+        present_health_kwargs = set(args) & health_kwargs
+        no_healthcheck = args.pop("no_healthcheck", None)
+
+        if no_healthcheck and present_health_kwargs:
+            raise ValueError(
+                f"Cannot set {sorted(present_health_kwargs)[0]} when no_healthcheck is True"
```
Comment (Honny1) [inline podman/domain/containers_create.py]: Non-blocking: I don't like indexing here. But it is okay.

File: podman/tests/integration/test_container_create.py [hunk 4]
```diff
@@ -262,6 +262,39 @@ def test_container_healthchecks(self):
         container = self.client.containers.create(self.alpine_image, **parameters)
         self.containers.append(container)
 
+    def test_container_healthcheck_fine_grained_kwargs(self):
```
Comment (Honny1) [inline podman/tests/integration/test_container_create.py:265]: I would also add a test for default values: `health_on_failure="none"`.

**Review by Honny1 — COMMENTED** [review]
I have some comments. Moust of them are small improvments.

File: podman/api/parse_utils.py (hunk 1 above)
Comment (inknos) [reply podman/api/parse_utils.py]: I was thinking of "1.5s" being valid but we could do integers only. I see the validation is wrong anyway so doing ints seems ok. WDYT?

File: podman/api/parse_utils.py (hunk 2 above)
Comment (inknos) [reply podman/api/parse_utils.py:35]: nice I'll fix it

File: podman/tests/integration/test_container_create.py (hunk 4 above)
Comment (inknos) [reply podman/tests/integration/test_container_create.py:265]: right

**Review by Honny1 — APPROVED** [review]
LGTM

**Comment by Honny1** [comment]
@inknos, can you please try to rebase on upstream main? CI failure seems to be unrelated.
